File: baekjoon/18406_lucky_straight/main.py
def solution(val):
    input_len = len(val)
    half = int(input_len / 2)
    
    a = 0
    b = 0

    for i in range(half):
        a = a + int(val[i])
        b = b + int(val[-1-i])

    if a == b:
        return "LUCKY"

    return "READY"

def input_test():
    val = input()
    print(solution(val))

def auto_test():
    tc = [
        ["123402", "LUCKY"],
        ["7755", "READY"],
        ["1234321", "LUCKY"]
    ]

    for i in range(len(tc)):
        res = solution(tc[i][0])
        print("TC {} : ".format(i))
        print("input : {}".format(tc[i][0]))
        print("output : {}".format(tc[i][1]))
        if res == tc[i][1]: print("PASS!")
        else: print("FAIL!")
        print()

input_test()
#auto_test()

File: baekjoon/18406_lucky_straight/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char* argv[]) {
	char input[16];
	int i;
	int sum1 = 0, sum2 = 0;
	int half;
	
	scanf("%s", input);
	half = strlen(input) / 2;
	
	for (i=0; i<half; i++) {
		sum1 += input[i] & 0x0f;
		sum2 += input[i+half] & 0x0f;
	}
	
	if (sum1 == sum2)
		printf("LUCKY\n");
	else
		printf("READY\n");
	
	return 0;
}